keras/src/layers/reshaping/reshape.py

```diff
@@ -11,11 +11,13 @@ class Reshape(Layer):
 
     Args:
         target_shape: Target shape. Tuple of integers, does not include the
-            samples dimension (batch size).
+            samples dimension (batch size). One element of the target_shape can
+            be -1 in which case the missing value is inferred from the length
+            of the array and remaining dimensions.
 
     Input shape:
-        Arbitrary, although all dimensions in the input shape must be
-        known/fixed. Use the keyword argument `input_shape` (tuple of integers,
+        Arbitrary, but required to be compatible with target shape.
+        Use the keyword argument `input_shape` (tuple of integers,
```

Comment:
>  Use the keyword argument `input_shape` ...

Now I'm noticing this documentation. This is actually a deprecated way of doing this. Plus, it's not needed.

Just remove that last part.